src/main/java/ch/wisv/areafiftylan/service/SeatServiceImpl.java

```diff
@@ -1,43 +1,124 @@
+import ch.wisv.areafiftylan.exception.TeamNotFoundException;
 import ch.wisv.areafiftylan.model.Seat;
+import ch.wisv.areafiftylan.model.Team;
+import ch.wisv.areafiftylan.model.Ticket;
 import ch.wisv.areafiftylan.model.User;
-import ch.wisv.areafiftylan.model.util.Coordinate;
-import ch.wisv.areafiftylan.service.repository.SeatRepository;
+import ch.wisv.areafiftylan.service.repository.SeatRespository;
+import ch.wisv.areafiftylan.service.repository.TicketRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import java.util.List;
+import java.util.*;
 
 @Service
 public class SeatServiceImpl implements SeatService {
 
-    SeatRepository seatRepository;
+    SeatRespository seatRespository;
+    TicketRepository ticketRepository;
+    TeamService teamService;
 
     @Autowired
-    public SeatServiceImpl(SeatRepository seatRepository) {
-        this.seatRepository = seatRepository;
+    public SeatServiceImpl(SeatRespository seatRespository, TicketRepository ticketRepository,
+                           TeamService teamService) {
+        this.seatRespository = seatRespository;
+        this.ticketRepository = ticketRepository;
+        this.teamService = teamService;
     }
 
     @Override
-    public Seat getSeatByUser(User user) {
-        return this.seatRepository.findByUser(user);
+    public Seat getSeatByUsername(String username) {
+        return seatRespository.findByTicketOwnerUsername(username)
+                .orElseThrow(() -> new SeatNotFoundException("User " + username + " doesn't have a seat"));
     }
 
     @Override
-    public List<Seat> getAllSeats() {
-        return seatRepository.findAll();
+    public SeatmapResponse getAllSeats() {
+        List<Seat> all = seatRespository.findAll();
+
+        Map<String, List<Seat>> seatMapResponse = new HashMap<>();
+
+        for (Seat seat : all) {
+            if (seatMapResponse.containsKey(seat.getSeatGroup())) {
+                seatMapResponse.get(seat.getSeatGroup()).add(seat);
+            } else {
+                List<Seat> seats = new ArrayList<>();
+                seats.add(seat);
+                seatMapResponse.put(seat.getSeatGroup(), seats);
+            }
+        }
+
+        return new SeatmapResponse(seatMapResponse);
+    }
+
+    @Override
+    public SeatmapResponse getSeatGroupByName(String groupname) {
+        List<Seat> seatGroup = seatRespository.findBySeatGroup(groupname);
+
+        Map<String, List<Seat>> seatMapResponse = new HashMap<>();
+        seatMapResponse.put(groupname, seatGroup);
+
+        return new SeatmapResponse(seatMapResponse);
+    }
+
+    @Override
+    public boolean reserveSeatForTicket(String groupname, int seatnumber, Long ticketId) {
+        Ticket ticket = ticketRepository.findOne(ticketId);
+        Seat seat = seatRespository.findBySeatGroupAndSeatNumber(groupname, seatnumber);
+        if (!seat.isTaken()) {
```

Comment:
Not sure if services in Spring are singletons by default. Otherwise, you need to synchronise manually on a singleton/static lock object. (Which is also a better practise over using the synchronized keyword)